Conversation View improvements

n-gao opened the pull request:
<!--
Thanks for contributing to the project!
Please help us keep this project in good shape by going through this checklist.
Replace the empty checkboxes [ ] below with checked ones [X] as they are completed
Remember, you can preview this before saving it.
-->

<!-- You can remove this first section if you have contributed before -->

### First time contributor checklist:

- [x] I have read the [README](https://github.com/signalapp/Signal-Desktop/blob/master/README.md) and [Contributor Guidelines](https://github.com/signalapp/Signal-Desktop/blob/master/CONTRIBUTING.md)
- [x] I have signed the [Contributor Licence Agreement](https://signal.org/cla/)

### Contributor checklist:

- [x] My contribution is **not** related to translations. _Please submit translation changes via our [Signal Desktop Transifex project](https://www.transifex.com/signalapp/signal-desktop/)._
- [x] My commits are in nice logical chunks with [good commit messages](http://chris.beams.io/posts/git-commit/)
- [x] My changes are [rebased](https://medium.freecodecamp.org/git-rebase-and-the-golden-rule-explained-70715eccc372) on the latest [`development`](https://github.com/signalapp/Signal-Desktop/tree/development) branch
- [x] A `yarn ready` run passes successfully ([more about tests here](https://github.com/signalapp/Signal-Desktop/blob/master/CONTRIBUTING.md#tests))
- [x] My changes are ready to be shipped to users

### Description
This PR intends to realize the intended changes by #4829. Thanks, @misterPaul0 for your great starting point.
I opened this PR because I think it is a neater implementation with less bug potential. However, a more experienced developer may correct me.
![image](https://user-images.githubusercontent.com/9084670/106984694-5b3e4780-6768-11eb-8230-f2cd049a8988.png)
![image](https://user-images.githubusercontent.com/9084670/106984816-8cb71300-6768-11eb-915d-94c36cfeae88.png)


These are the changes:
* Consecutive messages from the same author have less space in between and corner radii are adjusted to group them.
* Only a single profile picture is shown per group of messages (in a group chat)

Compared to the original PR (#4829) I did not include any changes to the timestamps. Also, all attachments should work just fine.

A question to the core team would still be whether there is a neater way to access the previous and next message than accessing the conversation via `window.getConversations`.
<!--
Describe briefly what your pull request changes. Focus on the value provided to users.

Does it address any outstanding issues in this project?
  https://github.com/signalapp/Signal-Desktop/issues?utf8=%E2%9C%93&q=is%3Aissue
  Reference an issue with the hash symbol: "#222"
  If you're fixing it, use something like "Fixes #222"

Please write a summary of your test approach:
  - What kind of manual testing did you do?
  - Did you write any new tests?
  - What operating systems did you test with? (please use specific versions: http://whatsmyos.com/)
  - What other devices did you test with? (other Desktop devices, Android, Android Simulator, iOS, iOS Simulator)
-->

n-gao commented:
I can not figure out why the CI is failing. The errors seem completely unrelated to my changes and the tests do not fail on my local windows machine.

hiqua commented:
Can you rewrite your commit history (maybe squash all commits into one)? It'll make it easier for others to try and fix the error. Right now it doesn't qualify for `My commits are in nice logical chunks with good commit messages`.

GitMatze commented:
Is this one larger gap in your second demo image intended?

n-gao commented:
@hiqua I squashed the previous commits.
@GitMatze Good catch, this was an issue with new messages as the height of the previous message has to be recomputed. I fixed this and it does not occur with new messages.

EvanHahn-Signal commented:
Sorry for the lack of updates here.

We have some intention of implementing this feature and have started to build the infrastructure to make it possible. I think we're going to implement this ourselves, but we will refer to your hard work (and the work of similar PRs) as a template.

I'm going to close this issue because we won't merge it as is, but again, thank you for your efforts.
